### PR TITLE
chore: Cleanup and fixes

### PR DESCRIPTION
**Pull request by xoscar**
This PR adds a bunch of cleanup changes to make it easier to test, including upgrading the OTEL dependencies

## Changes

- Updates the otel dependencies
- Upgrades the tracetest dependencies
- Adds scripts to test k6 and all types
- Updates scripts to match latest tracetest npm pacakges
- Adds tail_sampling to the otel collector

## Fixes

- https://github.com/kubeshop/tracetest-cloud/issues/899

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: collector.config.yaml [hunk 1]
```diff
@@ -11,11 +11,24 @@ receivers:
 processors:
   batch:
 
+  tail_sampling:
```
Comment (jorgeepc) [inline collector.config.yaml:14]: Shouldn't we use the filter approach instead?

File: collector.config.yaml [hunk 2]
```diff
@@ -11,11 +11,24 @@ receivers:
...
+        },
+      ]
+
 exporters:
   logging:
     loglevel: debug
-  jaeger:
-    endpoint: ${JAEGER_ENDPOINT}
+  otlp:
+    endpoint: jaeger:4317
```
Comment (jorgeepc) [inline collector.config.yaml:31]: Do we want to hardcode this?

File: collector.config.yaml (hunk 1 above)
Comment (xoscar) [reply collector.config.yaml:14]: I think the trace state policy is fine but maybe @mathnogueira has a different opinion

File: collector.config.yaml (hunk 2 above)
Comment (xoscar) [reply collector.config.yaml:31]: Yeah its all dockerized

File: collector.config.yaml (hunk 1 above)
Comment (mathnogueira) [reply collector.config.yaml:14]: the filter preprocessor is faster and doesn't require as much memory as this one. I mean, it's ok to use this here, but the filter would be better.

**Review by mathnogueira — APPROVED** [review]

File: collector.config.yaml (hunk 1 above)
Comment (xoscar) [reply collector.config.yaml]: Just changed it to use filter instead, seems to be working correctly!

**Pull request merged**